### ISSUE TITLE
prefer-matches should be prefer-isMatch

### ISSUE THREAD

**Issue by JnJSW** [wix/eslint-plugin-lodash#77]
The prefer-matches rule does not work as currently constructed.    The function [_.matches](https://lodash.com/docs#matches) only takes one argument and returns a function.  Whereas [._isMatch](https://lodash.com/docs#isMatch) will take two arguments and return a boolean.  I'm pretty sure that's what you're looking for.

Thanks so much for this terrific library.

**Comment by mik01aj** [comment]
I think this is caused by my mistake when I originally proposed this rule in #4. 😊

**Comment by ganimomer** [comment]
Huh, I thought maybe `_.isMatch` was introduced after this rule but it seems that it's been there since version 3.
That does seem like a better fit (except, since ESLint says rules should be kebab case, it should be `prefer-is-match`, like our rule `prefer-is-nil`).
And I think the most important part to change would be the error message, not the rule name.

**Comment by ganimomer** [comment]
I only changed the error message for now and not the rule name, since removing a rule is a breaking change (anyone who uses the rule explicitly and not in a config would have to change it, or ESLint would fail).

**Comment by jfmengels** [comment]
Migration path idea: You could rename the rule, and then add a new entry in the `index.js` file with the old name to link to the new file. That way people can migrate easily, and you can deprecate `prefer-matches` early without removing it.

It's a bit trickier in this repo since you automated the indexing, but doable nonetheless

```js
// index.js
var rules = _.assign(
  _.zipObject(rules, rules.map(rule => require(`./rules/${rule}`))),
  {'prefer-matches': require('./rules/prefer-isMatch')}
);
module.exports = {
  rules: rules,
}
```

**Comment by ganimomer** [comment]
Thanks for the offer, but that would mean that the rule would exist twice in the meantime, and every time a user violates it, they would get two errors with the same message per violation.

I think the best course of action is to leave this for now and put it on the list of things to change in the next major version.
